homeassistant/components/system_bridge/config_flow.py

```diff
@@ -114,12 +113,100 @@ class SystemBridgeConfigFlow(
     """Handle a config flow for System Bridge."""
 
     VERSION = 1
+    MINOR_VERSION = 2
 
     def __init__(self) -> None:
         """Initialize flow."""
         self._name: str | None = None
         self._input: dict[str, Any] = {}
         self._reauth = False
+        self._system_data: System | None = None
+
+    async def _validate_input(
+        self,
+        data: dict[str, Any],
+    ) -> dict[str, str]:
+        """Validate the user input allows us to connect.
+
+        Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+        """
+        host = data[CONF_HOST]
+
+        websocket_client = WebSocketClient(
+            host,
+            data[CONF_PORT],
+            data[CONF_TOKEN],
+        )
+
+        async def async_handle_module(
+            module_name: str,
+            module: Any,
+        ) -> None:
+            """Handle data from the WebSocket client."""
+            _LOGGER.debug("Set new data for: %s", module_name)
+            if module_name == "system":
+                self._system_data = module
+
+        try:
+            async with asyncio.timeout(15):
+                await websocket_client.connect(
+                    session=async_get_clientsession(self.hass)
+                )
+                self.hass.async_create_task(
+                    websocket_client.listen(callback=async_handle_module)
```

Comment:
Where do we close the websocket client and clean up the listen task?

Reply:
This is done here:

[async def close_websocket(_) -> None:](https://github.com/timmo001/core/blob/917f0f849db36c5b8644f8d600299d4284e56a0e/homeassistant/components/system_bridge/coordinator.py#L241-L249)

I can add additional cleanup to this function

Reply:
We don't use the coordinator in the config flow, right? The config flow should be isolated and clean up after itself.